File: DemoServer/PythonDemo/demo_methods/Basic/Edit.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Edit(object):
    @staticmethod
    def edit(company_id=1):
        with DocumentStoreHolder.get_store().open_session() as session:
            company = session.load("companies/" + company_id, object_type=Company)

            company.Address.Line2 = "Zip 12345"
            session.save_changes()

            return "CompanyId {0} Edited Successfully.".format(company_id)

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/basic/Edit.java
package net.ravendb.demo.basic;

import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class Edit {

    @GetMapping("/basic/edit")
    public String edit(
            @RequestParam (value = "companyId", defaultValue = "companies/1-A") String companyId) {

        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {

            Company company = session.load(Company.class, companyId);
            company.getAddress().setLine2("Zip 12345");
            session.saveChanges();

            return String.format("%s Edited Successfully", companyId);
        }
    }
}
